www/experiments/findings.inc

```diff
@@ -23,86 +24,137 @@ if ($experiment) {
 
 
     $diffMetrics = array(
-        //array( "TTFB", "Time to First Byte", "s"),
-        array( "StartRender", "Start Render", "s"),
-        array( "firstContentfulPaint", "First Contentful Paint", "s"),
-        array( "SpeedIndex", "Speed Index", "s"),
-        array( "chromeUserTiming.LargestContentfulPaint", "Largest Contentful Paint", "s"),
-        array( "chromeUserTiming.CumulativeLayoutShift", "Cumulative Layout Shift", "", array("Better","Worse")),
-        array( "TotalBlockingTime", "Total Blocking Time", "s"),
-        array( "visualComplete", "Visual Complete", "s"),
-        array( "FullyLoaded", "Fully Loaded", "s"),
-        array( "BytesIn", "Total Bytes", "kb", array("Lighter","Heavier"))
+        //array( "TTFB", "Time to First Byte", "s"), 
+        array("render", "Start Render", "s"),
+        array("firstContentfulPaint", "First Contentful Paint", "s"),
+        array("SpeedIndex", "Speed Index", "s"),
+        array("chromeUserTiming.LargestContentfulPaint", "Largest Contentful Paint", "s"),
+        array("chromeUserTiming.CumulativeLayoutShift", "Cumulative Layout Shift", "", array("Better", "Worse")),
+        array("TotalBlockingTime", "Total Blocking Time", "s"),
+        array("visualComplete", "Visual Complete", "s"),
+        array("fullyLoaded", "Fully Loaded", "s"),
+        array("BytesIn", "Total Bytes", "kb", array("Lighter", "Heavier"))
     );
 
-    ?>
-<div class="scrollableTable">
-    <table id="tableResults" class="pretty">
-        <tbody>
-            <tr class="metric_labels">
-                <?php
-                foreach ($diffMetrics as $metric) {
-                    $experimentMetric = $tests[0]['stepResult']->getMetric($metric[0]);
-                    $thisDiff = $tests[1]['stepResult']->getMetric($metric[0]) - $experimentMetric;
-                    $thisDiff = floatval(removeLeadingZero(round($thisDiff, 3)));
-                    if (abs($thisDiff) > 0) {
-                        echo '<th align="center" valign="middle">' . $metric[1] . '</th>';
+
+    // see if there's a notable diff in TTFB first, which is usually unrelated to the experiment
+    // offer a link and query string to ignore ttfb in some metrics
+    $experimentTTFB = $tests[0]['stepResult']->getMetric("TTFB");
+    $controlTTFB = $tests[1]['stepResult']->getMetric("TTFB");
+    $ttfbDiff = $tests[1]['stepResult']->getMetric("TTFB") - $tests[0]['stepResult']->getMetric("TTFB");
+    $ttfbDiffClean = floatval(removeLeadingZero(round($ttfbDiff, 3)));
+    $ttfbDiffClean = abs(round($ttfbDiffClean / 1000, 2));
+    $ttfbDiffNotable = abs($ttfbDiff) > 100;
+    $removeTTFBfromDiffs = isset($_GET['ignoreTTFB']);
+
+
+    // check if metric might benefit from TTFB removal
+    function ignoreTTFBforMetric($metric)
+    {
+        global $removeTTFBfromDiffs;
+        if (
+            $removeTTFBfromDiffs
+            && ($metric === "render"
```

Comment:
Instead of having to list out metric names again, could we append a value to the $diffMetrics array indicating whether or not a metric should be impacted by TTFB removal? Just thinking it avoids duplicating metric names.

Reply:
sure! good call.

Reply:
yeah, we need to move all these strings as const/struct of some sort at some point. String are too easy to mistype

Reply:
done